livekit-agents/livekit/agents/stt/stt.py

```diff
@@ -2,26 +2,29 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass, field
-from enum import Enum
+from enum import Enum, unique
 from typing import AsyncIterator, List
 
 from livekit import rtc
 
 from ..utils import AudioBuffer
 
 
+@unique
 class SpeechEventType(Enum):
-    # indicate the start of speech
-    # if the STT doesn't support this event, this will be emitted as the same time as the first INTERIM_TRANSCRIPT
-    START_OF_SPEECH = 0
-    # interim transcript, useful for real-time transcription
-    INTERIM_TRANSCRIPT = 1
-    # final transcript, emitted when the STT is confident enough that a certain
-    # portion of speech will not change
-    FINAL_TRANSCRIPT = 2
-    # indicate the end of speech, emitted when the user stops speaking
-    # the first alternative is a combination of all the previous FINAL_TRANSCRIPT events
-    END_OF_SPEECH = 3
+    START_OF_SPEECH = "start_of_speech"
+    """indicate the start of speech
+    if the STT doesn't support this event, this will be emitted as the same time as the first INTERIM_TRANSCRIPT"""
+    INTERIM_TRANSCRIPT = "interim_transcript"
+    """interim transcript, useful for real-time transcription"""
+    FINAL_TRANSCRIPT = "final_transcript"
+    """inal transcript, emitted when the STT is confident enough that a certain
+    portion of speech will not change"""
+    END_OF_SPEECH = "end_of_speech"
+    """indicate the end of speech, emitted when the user stops speaking
+    the first alternative is a combination of all the previous FINAL_TRANSCRIPT events"""
+    END_OF_UTTERANCE = "end_of_utterance"
```

Comment:
what's the difference between `end_of_utterance` and `end_of_speech`? From DeepGram's docs, it seems like they are the same thing: https://developers.deepgram.com/docs/utterance-end

Reply:
Deepgram doesn't have end_of_speech (We simulate it inside our integration) using the final_transcripts (endpoints). The endpoints are based on a VAD

end of utterance isn't based on VAD. So even if there is a lot of noise or inaccuracy on their VAD. It'll still be fired

Reply:
end_of_speech is here for consistency with other STTs

Reply:
got it, so what is the benefit of adding another event? vs using the current approach of translating end of utterance to `end_of_speech` event?

Reply:
end_of_speech event will certainly be called sooner than end_of_utterance as the STT may do more checks (wait more or some NLP based detection to see if a sentence is finished)